.github/workflows/wheels.yml

```diff
@@ -0,0 +1,81 @@
+name: Wheels
+
+on:
+  pull_request:  #TODO remove this line
+  release:
+    types:
+      - published
+
+env:
+  CIBW_TEST_COMMAND: pytest {project}/tests
+  CIBW_TEST_EXTRAS: test
+
+
+jobs:
+  build_sdist:
+    name: Build SDist
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+        fetch-depth: 0  # history required so cmake can determine version
+
+    - name: Build SDist
+      run: pipx run build --sdist
+
+    - name: Check metadata
+      run: pipx run twine check dist/*
+
+    - uses: actions/upload-artifact@v2
+      with:
+        path: dist/*.tar.gz
+
+
+  build_wheels:
+    name: Wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+        fetch-depth: 0  # history required so cmake can determine version
+
+    - uses: pypa/cibuildwheel@v2.3.0
+      env:
+        MACOSX_DEPLOYMENT_TARGET: "10.15"
+
+    - name: Verify clean directory
+      run: git diff --exit-code
+      shell: bash
+
+    - name: Upload wheels
+      uses: actions/upload-artifact@v2
+      with:
+        path: wheelhouse/*.whl
+
+
+  upload_all:
+    name: Upload if release
+    needs: [build_wheels, build_sdist]
+    runs-on: ubuntu-latest
+    #TODO uncomment this line
+    #if: github.event_name == 'release' && github.event.action == 'published'
```

Comment:
This is triggered when you publish a release on github. I am honestly not sure what is better, but I am thinking of switching also the conda package build to trigger from releases: As it is now you can create a tag (without making a release) to trigger a conda release build. Is that desirable?

Reply:
Does the tag have to match the pattern of a version number? If not, I think this approach here is better because it allows tagging for other reasons. An in general, tying conda / pip releases to GH releases sounds cleaner to me.

Reply:
We trigger release builds only from tags matching *.*.*